python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,581 @@
+
+        self.measurement.run(sources, exposure)
+        self.normalized_calibration_flux.run(exposure=exposure, catalog=sources)
+        self.apply_aperture_correction.run(sources, exposure.apCorrMap)
+        self.catalog_calculation.run(sources)
+        self.set_primary_flags.run(sources)
+
+        sources["visit"] = exposure.visitInfo.id
+        sources["detector"] = exposure.info.getDetector().getId()
+
+        self._match_calib_sources(sources, calib_sources, exposure.info.getDetector().getId())
+
+        return sources
+
+    def _match_calib_sources(self, sources, calib_sources, detector):
+        """Match with calib_sources to set `calib_*` flags in the output
+        catalog.
+
+        Parameters
+        ----------
+        sources : `lsst.afw.table.SourceCatalog`
+            Catalog that was detected and measured on the exposure. Modified
+            in place to set the psf_fields.
+        calib_sources : `astropy.table.Table`
+            Per-visit catalog of measurements to get 'calib_*' flags from.
+        detector : `int`
+            Id of detector for this exposure, to get the correct sources from
+            calib_sources for cross-matching.
+        """
+        # NOTE: we don't remove the sky sources here, but they should be very
+        # far from any actual source, and so should not match with anything.
+        use = calib_sources["detector"] == detector
+        with smatch.Matcher(sources["coord_ra"], sources["coord_dec"]) as matcher:
+            _, i1, i2, _ = matcher.query_knn(
+                calib_sources[use]["coord_ra"],
+                calib_sources[use]["coord_dec"],
+                1,
+                self.config.calib_match_radius / 3600.0,
+                return_indices=True,
+            )
+
+        for field in self.psf_fields:
+            # NOTE: Have to fill a full-sized array first, then set with it.
+            result = np.zeros(len(sources), dtype=bool)
+            result[i1] = calib_sources[use][i2][field]
+            sources[field] = result
+
+    def _apply_photo_calib(self, exposure, sources_footprints, photo_calib):
+        """Photometrically calibrate the exposure and catalog with the
+        supplied PhotoCalib, and set the exposure's PhotoCalib to 1.
+
+        Parameters
+        ----------
+        exposures : `lsst.afw.image.Exposure`
+            Exposure to calibrate and set PhotoCalib on; Modified in place.
+        sources_footprints : `lsst.afw.table.SourceCatalog`
+            Catalog to calibrate.
+        photo_calib : `lsst.afw.image.PhotoCalib`
+            Photometric calibration to apply.
+        calibrated_stars : `lsst.afw.table.SourceCatalog`
+            Star catalog with flux/magnitude columns computed from the
+            supplied PhotoCalib.
+        """
+        sources_footprints = photo_calib.calibrateCatalog(sources_footprints)
+        # TODO: This is temporary, until we can do the measurements on the
```

Comment:
Another bare TODO?

Reply:
This is like the other one I noted above, where it's such a nebulous todo ("make our measurement task aware that the exposure is calibrated, so it can measure directly in njy flux units instead of instFlux") with such a long time horizon that I've removed the TODO, but left the note for future reference.